### PR TITLE
Setting new texture on sprite will reset the program state that is changed

### PR DESCRIPTION
**Pull request by Y-way**
Setting new texture on sprite will reset the ProgramState that is seted user`s program state

### CODE REVIEW

File: cocos/2d/CCSprite.cpp [hunk 1]
```diff
@@ -372,8 +372,10 @@ void Sprite::setTexture(const std::string &filename)
 
 void Sprite::setTexture(Texture2D *texture)
 {
-    setGLProgramState(GLProgramState::getOrCreateWithGLProgramName(GLProgram::SHADER_NAME_POSITION_TEXTURE_COLOR_NO_MVP, texture));
-
+    if(_glProgramState == nullptr)
+    {
+        setGLProgramState(GLProgramState::getOrCreateWithGLProgramName(GLProgram::SHADER_NAME_POSITION_TEXTURE_COLOR_NO_MVP, texture));
```
Comment (minggo) [inline cocos/2d/CCSprite.cpp:377]: GLProgramState::getOrCreateWithGLProgramName will return the same glprograme state, why it will be changed?

Comment (Y-way) [reply cocos/2d/CCSprite.cpp:377]: In my project:
1.set default texture.
2.set mine shader program
3.set new texture.

then, the sprite will reset my shader program.

Comment (minggo) [reply cocos/2d/CCSprite.cpp:377]: sounds reasonable

**Comment by minggo** [comment]
Please change the commit message to english or send a new PR. Thanks.

**Pull request closed without merging**